### PR TITLE
New Fix on KFParticle Magnetic Field using the Official Map

### PR DESCRIPTION
**Pull request by MYOMAO**
We identified that unit and the direction are wrong in the official KFParticle codes. We made a pull request [#1987 ](url) before to fix this error. Somehow we did not do the commit correctly and have to close the pull request. Now we make sure we do all github commands correctly and open a fresh pull request. In this pull request, we take the suggestions from Chris and Joe to use the official full file name from the CDB and load the magnetic field map for KFParticle.  

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

[comment]: <> ( What does this PR do? Linking to talk in software meeting encouraged )

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

### CODE REVIEW

**Comment by blackcathj** [comment]
Last report was github connectivity issue. Restarting

File: offline/packages/KFParticle_sPHENIX/KFParticle_sPHENIX.cc [hunk 1]
```diff
@@ -34,12 +34,16 @@
...
-#include <map>       // for map
-#include <tuple>     // for tie, tuple
+#include <ffamodules/CDBInterface.h>  // for accessing the field map file from the CDB
+#include <cctype>                     // for toupper
+#include <cmath>                      // for sqrt
+#include <cstdlib>                    // for size_t, exit
+#include <iostream>                   // for operator<<, endl, basi...
+#include <map>                        // for map
+#include <tuple>                      // for tie, tuple
+#include "TFile.h"                    // for reading the field map file
```
Comment (osbornjd) [inline offline/packages/KFParticle_sPHENIX/KFParticle_sPHENIX.cc:44]: You need to use <> for these ROOT includes and not "" as Chris mentioned

File: offline/packages/KFParticle_sPHENIX/KFParticle_sPHENIX.cc [hunk 2]
```diff
@@ -91,6 +95,36 @@ int KFParticle_sPHENIX::Init(PHCompositeNode *topNode)
...
 
+  // Load the official offline B-field map that is also used in tracking, basically copying the codes from: https://github.com/sPHENIX-Collaboration/coresoftware/blob/master/offline/packages/trackreco/MakeActsGeometry.cc#L478-L483, provide by Joe Osborn.
+  char *calibrationsroot = getenv("CALIBRATIONROOT");
+  std::string m_magField = "sphenix3dtrackingmapxyz.root";
+  // std::string url = CDBInterface::instance()->getUrl("FIELDMAPTRACKING", m_magField);
+  m_magField = CDBInterface::instance()->getUrl("FIELDMAPTRACKING");  // Joe's Implementation to get the field map file name
+
+  if (calibrationsroot != nullptr)
+  {
+    m_magField = std::string(calibrationsroot) + std::string("/Field/Map/") + m_magField;
```
Comment (osbornjd) [inline offline/packages/KFParticle_sPHENIX/KFParticle_sPHENIX.cc:106]: I gave you bad information, I tested this locally just now and it works
```
 char *calibrationsroot = getenv("CALIBRATIONROOT");
m_magField = "sphenix3dtrackingmapxyz.root";
      
 if (calibrationsroot != nullptr)
   {
    m_magField = std::string(calibrationsroot) + std::string("/Field/Map/") + m_magField;
  }
      
      m_magField = CDBInterface::instance()->getUrl("FIELDMAPTRACKING", m_magField);
```
Please use this and not what you currently have. Sorry for my mistake

**Review by osbornjd — CHANGES_REQUESTED** [review]

**Comment by MYOMAO** [comment]
OK I am fixing it now and resubmit my commit. Thank you very much!

**Comment by MYOMAO** [comment]
Again, I messed up this pull request. I will again delete this one and create a new pull request.

**Pull request closed without merging**